subsys/logging/backends/log_backend_net.c

```diff
@@ -196,20 +197,31 @@ static int format_set(const struct log_backend *const backend, uint32_t log_type
 	return 0;
 }
 
+bool log_backend_net_set_addr(const char *addr)
+{
+	net_sin(&server_addr)->sin_port = htons(514);
+
+	bool ret = net_ipaddr_parse(addr, strlen(addr), &server_addr);
```

Comment:
The thing with runtime address configuration is that with the current implementation it will have no effect if the corresponding net_context has already been initialized (see `do_net_init()` and `net_context_connect()` in there).

I think that at the very least, the function should return an error in such case (`net_init_done` is set). Perhaps a better solution though would be to release the current `net_context` and clear `net_init_done` flag so that the log backend can create a new one, with a new destination address. WDYT?

Reply:
It'll for sure make this feature much more flexible and adding flexibility was my initial motivation. I'll add a way to change the server address on the fly.